File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        def goodH(root, greatest):
            if not root:
                return 0
            maxVal = max(root.val, greatest)
            lAndR = goodH(root.left, maxVal) + goodH(root.right, maxVal)
            if root.val >= greatest:
                return 1 + lAndR
            return lAndR

        return goodH(root, -101)

File: Data Structures & Algorithms/balanced-binary-tree/submission-0.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

class Solution {
    public boolean isBalanced(TreeNode root) {
        if (root == null) {
            return true;
        }
        if (maxDepthH(root.left, 0) - maxDepthH(root.right, 0) == 1 || 
        maxDepthH(root.left, 0) - maxDepthH(root.right, 0) == 0 || maxDepthH(root.left, 0) - maxDepthH(root.right, 0) == -1) {
            return true && isBalanced(root.left) && isBalanced(root.right);
        }
        else {
            return false;
        }
    }

    public int maxDepthH(TreeNode current, int depth) {
        if (current == null) {
            return depth;
        }
        else {
            return Math.max(maxDepthH(current.left, depth + 1), maxDepthH(current.right, depth + 1));
        }
    }
}
